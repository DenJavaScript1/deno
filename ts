#!/usr/bin/env -S deno run
// deno-lint-ignore-file prefer-const
let message = 'Hello, Deno!';
console.log(message);