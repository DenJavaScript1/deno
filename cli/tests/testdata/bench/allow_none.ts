<<<<<<< HEAD
import { unreachable } from "../../../../test_util/std/assert/mod.ts";
=======
import { unreachable } from "../../../../test_util/std/testing/asserts.ts";
>>>>>>> 172e5f0a0 (1.38.5 (#21469))

const permissions: Deno.PermissionName[] = [
  "read",
  "write",
  "net",
  "env",
  "run",
  "ffi",
  "hrtime",
];

for (const name of permissions) {
  Deno.bench({
    name,
    permissions: {
      [name]: true,
    },
    fn() {
      unreachable();
    },
  });
}
