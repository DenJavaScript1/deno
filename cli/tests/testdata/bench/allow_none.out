[WILDCARD]
running 7 benches from [WILDCARD]
bench read ... FAILED [WILDCARD]
bench write ... FAILED [WILDCARD]
bench net ... FAILED [WILDCARD]
bench env ... FAILED [WILDCARD]
bench run ... FAILED [WILDCARD]
bench ffi ... FAILED [WILDCARD]
bench hrtime ... FAILED [WILDCARD]

failures:

read
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

write
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

net
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

env
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

run
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

ffi
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

hrtime
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

failures:

	read
	write
	net
	env
	run
	ffi
	hrtime

bench result: FAILED. 0 passed; 7 failed; 0 ignored; 0 measured; 0 filtered out [WILDCARD]
